src/AzureRenderManager/WebApp/Controllers/EnvironmentsController.cs

```diff
@@ -1305,11 +1339,15 @@ private async Task CreateOrUpdateStorageAndBatchAccounts(RenderingEnvironment en
                 };
             }
 
-            await _azureResourceProvider.CreateFilesShare(
-                environment.StorageAccount.SubscriptionId,
-                environment.StorageAccount.ResourceGroupName,
-                environment.StorageAccount.Name,
-                model.NewFileShareName);
+            if (model.CreateFilesShare)
+            {
+                await Task.Delay(10000);
```

Comment:
This seems a bit brute force... Is 10 seconds always enough or should this have retry logic similar to what you've added for registering the new provider?